### DISCUSSION TITLE
Can recovery codes always be used in place of mobile authentication?

### DISCUSSION THREAD

**Discussion by OverLordGoldDragon** [community/community#72253] [Other Feature Feedback, Questions, & Ideas] [answered]
After setting up mobile 2FA with a QR scan, I was given recovery codes, and have used them once. I can also generate new.

Can I always use recovery codes instead of having a code sent to mobile?

_Optional_: if "yes", what is the purpose of the first-time mobile setup?

**Comment by omarchouman** [comment]
Yes, you can use recovery codes instead of receiving a code on your mobile device every time you log in. The purpose of the first-time mobile setup with 2FA (Two-Factor Authentication) is to establish a secure and additional layer of authentication for your account.

When you set up 2FA with a QR scan on your mobile device, it associates your account with a unique code generated by a 2FA app (like Google Authenticator or Authy). This code changes periodically and is used in addition to your regular password when logging in. The idea is to add an extra layer of security to your account by requiring something you know (your password) and something you have (the mobile device with the 2FA app).

The recovery codes serve as a backup in case you lose access to your 2FA device. Each recovery code is a one-time use code that can be used to log in without needing the current code from your 2FA app. However, it's important to note that once a recovery code is used, it becomes invalid, and you may need to generate new recovery codes.

So, while you can use recovery codes to log in, the primary purpose of the mobile setup is to provide a convenient and secure way to generate time-sensitive codes for authentication, making it more difficult for unauthorized users to access your account even if they have your password. The recovery codes are a failsafe in case you lose access to your 2FA device.

**Reply by OverLordGoldDragon** [reply]
Thanks, but I prefer to hear from staff or see documentation. I've sent a support ticket in case they don't roam here.

Your explanation of necessity of mobile doesn't make sense to me. Code generation can be done by any device, I don't see how associating it with a specific device adds any security.

**Reply by omarchouman** [reply]
Appreciate your perspective. What I was trying to say is as much as it became a best practice it is kind of considered an "Essential" thing to have right now.

And for the reasons. 

You always find online and through articles these top 3 reasons that "make sense"

Limiting access to the code generation process: If code generation can only be done on a specific device, then only authorized users with access to that device will be able to generate code. This can help to prevent unauthorized access to code, which could be used for malicious purposes.

Tracking changes to the code: If code generation is associated with a specific device, then it is possible to track all changes to the code that is generated on that device. This can help to identify any unauthorized or malicious changes to the code.

Verifying the authenticity of the code: If code generation is associated with a specific device, then it is possible to verify the authenticity of the code that is generated on that device

Note that this is also opinionated and that was my opinion though.

Thank you

**Reply by OverLordGoldDragon** [reply]
I can already generate codes on any desktop I access. The only conceivable point would be associating some generation "seed" with a device, but I can't imagine how it'd be more secure than random generation. Generate from any device, enter into any device. What's device-specific is the mobile option, not recovery codes (and there of course it makes sense).

**Comment by OverLordGoldDragon** [comment, marked as answer]
Github support confirmed that recovery codes can always be used.

As for why require the first-time setup, I got same corporate lingo. They have no real technical reason. They just _really_ want people using mobile.

(There's semi-valid points on e.g. losing the recovery codes, but make of it what you will)

**Reply by nishandha** [reply]
[Uploading GoogleCybersecurityTimeline_Vertical_RGB.pdf…]()
